GTO docs (#199)
iterative/mlem.ai: content/docs/gto/user-guide.md

> # User Guide
> 
> GTO helps you build a Artifact Registry out of your Git repository. It does so
> by creating Git tags of special format and managing `artifacts.yaml` metafile.

Reviewer: 1/2

```suggestion
# User Guide

GTO helps you build an Artifact Registry out of your Git repository. It creates
annotated [Git tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging) with a
[special format](#git-tag-message-format) in their message, and manages an
`artifacts.yaml` file.
```

Since committing large files to Git is not a good practice, you should consider
not committing your artifacts to Git. Instead, use [DVC](https://dvc.org/doc),
Git-lfs, or any method commit pointers to the binary files instead.

> ## Annotations in artifacts.yaml
> 
> Using Git tag to register versions and assign stages is handy, but the Git tag
> itself doesn't contain path to the artifact, type of it (it could be `model` or
> `dataset`), or any other information you may find useful. For simple projects
> (e.g. single artifact) we can assume the details in a downstream system. But for
> more advanced cases, we should codify them in the registry itself.

Reviewer: ```suggestion
## Annotations in artifacts.yaml

Using Git tag to register versions and assign stages is handy, but the Git tag
itself doesn't contain path to the artifact, type of it (it could be `model` or
`dataset`), or any other information you may find useful. For simple projects
(e.g. single artifact) we can assume the details
[in a CI/CD system](#acting-in-ci-cd) downstream. But for more advanced cases,
we should codify them in the registry itself.
```

To keep this metainformation, GTO uses `artifacts.yaml` file. Commands like
`gto annotate` and `gto remove` are used to modify it, while `gto describe`
helps get them when they're needed.

If you would like to see an example of `artifacts.yaml`, check out the
[example-gto](https://github.com/iterative/example-gto/blob/main/artifacts.yaml)
repo.

> ## Acting downstream
> 
> Once Git tag was pushed, you can start acting downstream. One of the popular
> options here is to use CI/CD, setting it to be triggered by pushing a Git tag.

Reviewer: ```suggestion
## Acting in CI/CD

Once Git tags are pushed, you can start acting in systems downstream. A
popular options is to use CI/CD (triggered when Git tags are pushed).
```

The workflow uses [the GTO GH Action](https://github.com/iterative/gto-action)
that fetches all Git tags (to correctly interpret the Registry), finds out the
`version` of the artifact that was registered, the `stage` that was assigned,
and annotations details such as `path`, `type`, `description`, etc, so you could
use them in the next steps of the CI.

### Helpful commands

If you would like to set up CI/CD, but don't want to use GTO GH Action, check
out `gto show`, `gto check-ref` and `gto describe` commands.

## Configuring GTO

To configure GTO, use file `.gto` in the root of your repo:

```yaml
# .gto config file
types: [model, dataset] # list of allowed Types
stages: [dev, stage, prod] # list of allowed Stages
```

When allowed Stages or Types are specified, GTO will check commands you run and
error out if you provided a value that doesn't exist in the config. Note, that
GTO applies the config from the workspace, so if want to apply the config from
`main` branch, you need to check it out first with `git checkout main`.

Alternatively, you can use environment variables (note the `GTO_` prefix)

```cli
$ GTO_EMOJIS=false gto show
```

> ## Git tags format
> 
> <admon type="tip">
> 
> You can work with GTO without knowing the used Git tag names conventions, since
> commands like `gto register` and `gto assign` abstracts that away from you.

Reviewer: 2/2

```suggestion
## Git tag message format

<admon type="tip">

You can work with GTO without knowing these conventions, since
[`gto` commands](/doc/command-reference) take care of everything for you.
```

Reviewer: Were we able to update from `#git-tags-format` to `#git-tag-message-format` everywhere? 🙂

Author: Good catch. Overlooked the header's change here. It doesn't have anything with tag's message - it's only about tag names. So rolling back your changes on this, and fixing all links to this section in new PR.

Reviewer: > doesn't have anything (to do) with tag's message

Oh I see.
